tests/index.js

```diff
@@ -1,25 +1,134 @@
-var Repo = require('../repo')
 var Dat = require('..')
 var test = require('tape')
 
+var DAT_PROTOCOL = 'dat://'
+var DAT_KEY_STRING_LENGTH = 64
+var DAT_KEY_BYTE_LENGTH = 32
+var DAT_URL_LENGTH = DAT_PROTOCOL.length + DAT_KEY_STRING_LENGTH
+
 test('create a dat in memory', function (t) {
-  t.plan(7)
+  t.plan(6)
   var dat = new Dat()
   t.equals(dat.repos.length, 0, 'has zero repos before adding')
-  dat.add(function (repo) {
-    t.equals(dat.repos.length, 1, 'has one repo after adding')
-    t.ok(repo.key, 'repo ready and has a key')
-    t.equals(repo.key.length, 32, 'has key with proper length')
-    t.equals(repo.archive.key, repo.key, 'key is the archive key')
+  var repo = dat.add()
+
+  dat.on('repo', (repo) => {
+    t.ok(repo, 'emits the repo event')
+    t.equals(repo.url.length, DAT_URL_LENGTH, 'repo key is there')
+  })
+
+  t.equals(dat.repos.length, 1, 'has one repo after adding')
+  repo.ready(() => {
+    t.equals(repo.archive.key.length, DAT_KEY_BYTE_LENGTH, 'has key with proper length')
+    t.equals(repo.archive.key.toString('hex'), repo.url.slice(DAT_PROTOCOL.length), 'key is the archive key')
+
+    repo.close()
+
+    // TODO: What's this for?
     repo.on('close', function () {
+      dat.close()
     })
   })
-  dat.on('repo', function (repo) {
-    t.ok(repo, 'emits the repo event')
-    t.equals(repo.key.length, 32, 'repo key is there')
-    dat.close()
+
+  dat.on('close', () => {
+    t.end()
+  })
+})
+
+test('replicate a dat using WebRTC', function (t) {
+  var dat1 = new Dat()
+  var dat2 = new Dat()
+  t.equals(dat1.repos.length, 0, 'has zero repos before adding')
+  t.equals(dat2.repos.length, 0, 'has zero repos before adding')
+
+  var repo1 = dat1.add(null)
+
+  repo1.ready(() => {
+
+    repo1.archive.writeFile('/example.txt', 'Hello World!', (err) => {
+      t.notOk(err, 'no error when writing')
+
+      var url = repo1.url
+
+      repo2 = dat2.get(url)
+
+      repo2.archive.readFile('/example.txt', 'utf-8', (err, data) => {
+        t.notOk(err, 'no errors when reading')
+
+        t.equals(data, 'Hello World!', 'got proper data from archive')
+
+        dat1.close(() => {
+          dat2.close(() => {
```

Comment:
okay cool so these are getting tested that they're closing, disregard earlier comment :)